'list-tests: failed' still reporting successful tests.

See https://github.com/dorny/test-reporter/issues/142

I would be willing to contribute the fix described in the above issue. However, I am very new to type-script and have no idea how to get started.

@IanMoroney Would it be possible for you to provide information on how to build/run this so I can try/add the suggested fix? Thanks!

Looks like there's a linting problem in `get-report.ts`

As part of local development, if you run a `npm run all` it will include the `eslint` step.
I'll update the readme to include some local development steps 👍 

```
> eslint src/**/*.ts


/Users/work/Source/Repos/thirdparty/phoenix-test-reporting/src/report/get-report.ts
  235:17  error  Delete `;`       prettier/prettier
  235:17  error  Extra semicolon  semi
  235:17  error  Extra semicolon  @typescript-eslint/semi

✖ 3 problems (3 errors, 0 warnings)
  3 errors and 0 warnings potentially fixable with the `--fix` option.
  ```

--- a/__tests__/__outputs__/dotnet-trx-only-failed.md
+++ b/__tests__/__outputs__/dotnet-trx-only-failed.md
@@ -1,0 +1,25 @@
+![Tests failed](https://img.shields.io/badge/tests-5%20passed%2C%205%20failed%2C%201%20skipped-critical)
+## ❌ <a id="user-content-r0" href="#r0">fixtures/dotnet-trx.trx</a>
+**11** tests were completed in **1s** with **5** passed, **5** failed and **1** skipped.
+|Test suite|Passed|Failed|Skipped|Time|
+|:---|---:|---:|---:|---:|
+|[DotnetTests.XUnitTests.CalculatorTests](#r0s0)|5✔️|5❌|1✖️|118ms|
+### ❌ <a id="user-content-r0s0" href="#r0s0">DotnetTests.XUnitTests.CalculatorTests</a>
+```
+❌ Exception_In_TargetTest
+	System.DivideByZeroException : Attempted to divide by zero.
+❌ Exception_In_Test
+	System.Exception : Test
+❌ Failing_Test
+	Assert.Equal() Failure
+	Expected: 3
+	Actual:   2
+❌ Is_Even_Number(i: 3)
+	Assert.True() Failure
+	Expected: True
+	Actual:   False
+❌ Should be even number(i: 3)
+	Assert.True() Failure
+	Expected: True
+	Actual:   False
+```

--- a/__tests__/__outputs__/jest-test-results.md
+++ b/__tests__/__outputs__/jest-test-results.md
@@ -405,28 +405,11 @@
 |packages/pretty-format/src/__tests__/ReactElement.test.ts|3✔️|||64ms|
 ### ❌ <a id="user-content-r0s75" href="#r0s75">e2e/__tests__/jestChangedFiles.test.ts</a>
 ```
-✔️ gets hg SCM roots and dedupes them
-✔️ gets git SCM roots and dedupes them
-✔️ gets mixed git and hg SCM roots and dedupes them
-✔️ gets changed files for git
-✔️ monitors only root paths for git
-✔️ does not find changes in files with no diff, for git
-✔️ handles a bad revision for "changedSince", for git
 ❌ gets changed files for hg
 	Error: abort: empty revision range
-✔️ monitors only root paths for hg
-✔️ handles a bad revision for "changedSince", for hg
 ```
 ### ❌ <a id="user-content-r0s98" href="#r0s98">e2e/__tests__/onlyChanged.test.ts</a>
 ```
-✔️ run for "onlyChanged" and "changedSince"
-✔️ run only changed files
-✔️ report test coverage for only changed files
-✔️ report test coverage of source on test file change under only changed files
-✔️ do not pickup non-tested files when reporting coverage on only changed files
-✔️ collect test coverage when using onlyChanged
-✔️ onlyChanged in config is overwritten by --all or testPathPattern
 ❌ gets changed files for hg
 	Error: expect(received).toMatch(expected)
-✔️ path on Windows is case-insensitive
 ```

--- a/__tests__/dotnet-trx.test.ts
+++ b/__tests__/dotnet-trx.test.ts
@@ -3,7 +3,7 @@ import * as path from 'path'
 
 import {DotnetTrxParser} from '../src/parsers/dotnet-trx/dotnet-trx-parser'
 import {ParseOptions} from '../src/test-parser'
-import {getReport} from '../src/report/get-report'
+import {getReport, ReportOptions} from '../src/report/get-report'
 import {normalizeFilePath} from '../src/utils/path-utils'
 
 describe('dotnet-trx tests', () => {
@@ -40,6 +40,33 @@ describe('dotnet-trx tests', () => {
     expect(result).toMatchSnapshot()
 
     const report = getReport([result])
+    fs.mkdirSync(path.dirname(outputPath), {recursive: true})
+    fs.writeFileSync(outputPath, report)
+  })
+
+  it('matches report snapshot (only failed tests)', async () => {
+    const fixturePath = path.join(__dirname, 'fixtures', 'dotnet-trx.trx')
+    const outputPath = path.join(__dirname, '__outputs__', 'dotnet-trx-only-failed.md')
+    const filePath = normalizeFilePath(path.relative(__dirname, fixturePath))
+    const fileContent = fs.readFileSync(fixturePath, {encoding: 'utf8'})
+
+    const opts: ParseOptions = {
+      parseErrors: true,
+      trackedFiles: ['DotnetTests.Unit/Calculator.cs', 'DotnetTests.XUnitTests/CalculatorTests.cs']
+      //workDir: 'C:/Users/Michal/Workspace/dorny/test-check/reports/dotnet/'
+    }
+
+    const parser = new DotnetTrxParser(opts)
+    const result = await parser.parse(filePath, fileContent)
+    expect(result).toMatchSnapshot()
+
+    const reportOptions: ReportOptions = {
+      listSuites: 'all',
+      listTests: 'failed',
+      onlySummary: false,
+      baseUrl: ''
+    }
+    const report = getReport([result], reportOptions)
     fs.mkdirSync(path.dirname(outputPath), {recursive: true})
     fs.writeFileSync(outputPath, report)
   })

--- a/src/report/get-report.ts
+++ b/src/report/get-report.ts
@@ -231,6 +231,9 @@ function getTestsReport(ts: TestSuiteResult, runIndex: number, suiteIndex: numbe
     }
     const space = grp.name ? '  ' : ''
     for (const tc of grp.tests) {
+      if (options.listTests === 'failed' && tc.result !== 'failed') {
+        continue
+      }
       const result = getResultIcon(tc.result)
       sections.push(`${space}${result} ${tc.name}`)
       if (tc.error) {

